### PR TITLE
Adding changes in Variables_data_types slides, class exercises and homework

### PR DESCRIPTION
**Pull request by sfishbai**
Create changes in the Variables_data_types slides
Fix the doc of Variables_data_types Homework
Fix the doc of Variable_data_type class exercise

### CODE REVIEW

File: docs/course_materials/variables_data_types.html [hunk 1]
```diff
@@ -39,12 +39,75 @@
...
+                    <pre><code data-trim data-noescape class="hljs">
+    >>> name = 'Tom'
+    >>> family_name = 'Sawyer'
+    >>> print('Hello %s!' % name)
+    >>> Hello Tom!
+    >>> print('Hello %s %s!' % (name, family_name))
+    >>> Hello Tom Sawyer!
+                    </code></pre>
+                </section>
+                <section>
```
Comment (shiramax) [inline docs/course_materials/variables_data_types.html:80]: remove the section

File: docs/course_materials/variables_data_types.html [hunk 2]
```diff
@@ -39,12 +39,75 @@
...
+                <section data-markdown="exercises/03_Variables_and_datatypes/class_exercise_1.md" data-separator="^\n\n\n" data-separator-vertical="^\n\n">
+                </section>
+            </section>
```
Comment (shiramax) [inline docs/course_materials/variables_data_types.html:83]: same here

File: docs/course_materials/variables_data_types.html [hunk 3]
```diff
@@ -212,40 +295,11 @@ <h3><b>Examples</b></h3>
...
                     </code></pre>
                 </section>
-            </section>
-            <section>
-                <section>
-                    <h1><b>
-                            <span>Numbers</span>
-                    </h1></b>
-                </section>
                 <section>
```
Comment (shiramax) [inline docs/course_materials/variables_data_types.html:298]: remove

File: docs/course_materials/variables_data_types.html [hunk 4]
```diff
@@ -212,40 +295,11 @@ <h3><b>Examples</b></h3>
...
-            -2
-            >>> float(5)
-            5.0
-            >>> int("76")
-            76
-                    </code></pre>
+                <section data-markdown="exercises/03_Variables_and_datatypes/class_exercise_2.md" data-separator="^\n\n\n" data-separator-vertical="^\n\n">
                 </section>
             </section>
+            </section>
```
Comment (shiramax) [inline docs/course_materials/variables_data_types.html:302]: remove

File: docs/course_materials/exercises/03_Variables_and_datatypes/class_exercise_2.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+### Let's Practice Together \#2
+
+- Create a file named *variables_class.py*
+- Create a variable named x and assign the value 50 to it.
+- Display the sum of 5 + 10, using two variables: x and y.
+- Create a variable called z, assign x + y to it, and print the result.  
+        x = 5
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/class_exercise_2.md:7]: I think we can delete x =5 and y =10

File: docs/course_materials/exercises/03_Variables_and_datatypes/class_exercise_2.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+### Let's Practice Together \#2
+
+- Create a file named *variables_class.py*
+- Create a variable named x and assign the value 50 to it.
+- Display the sum of 5 + 10, using two variables: x and y.
+- Create a variable called z, assign x + y to it, and print the result.
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/class_exercise_2.md:6]:
```suggestion
- Create a variable  **z**, assign x + y to it, and print the result.  
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/class_exercise_2.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+### Let's Practice Together \#2
+
+- Create a file named *variables_class.py*
+- Create a variable named x and assign the value 50 to it.
+- Display the sum of 5 + 10, using two variables: x and y.
+- Create a variable called z, assign x + y to it, and print the result.  
+        x = 5  
+        y = 10
+- Create a variable named carname and assign the value "Volvo" to it.
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/class_exercise_2.md:9]:
```suggestion
- Create a variable  **carname** and assign the value "Volvo" to it.
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
+# Variables and Data Types - HomeWork exercise 
+
+## Variables, numbers and boolean 
+
+Before answering the questions:
+1. Change Directory to lesson4
+2. Create a directory called Homework
+3. Create a file called variables.py and answer the following:
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:8]:
```suggestion
3. Create a file  **variables.py** and answer the following:
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
+# Variables and Data Types - HomeWork exercise 
+
+## Variables, numbers and boolean 
+
+Before answering the questions:
+1. Change Directory to lesson4
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:6]:
```suggestion
1. Create a directory **lesson4** and change your current directory to **lesson4**
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+2. Create a directory called Homework
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:7]:
```suggestion
2. Create a directory  **homework** inside **lesson4**
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+3. Create a file called variables.py and answer the following:
+
+### variables_1.py - Exercise 1
+Add two variables together:
+1. Declare three variables (x, y, and z) as integers.
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:12]:
```suggestion
1. Declare three variables (**x**,** y**, and **z**) as integers.
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+2. Assign a value of 9 to x
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:13]:
```suggestion
2. Assign a value of 9 to **x**
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+3. Assign a value of 7 to y
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:14]:
```suggestion
3. Assign a value of 7 to **y**
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+4. Make z equal to the sum of those two variables.
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:15]:
```suggestion
4. Make **z** equal to the sum of those two variables. 
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+2. Create a directory called Homework
+3. Create a file called variables.py and answer the following:
+
+### variables_1.py - Exercise 1
+Add two variables together:
+1. Declare three variables (x, y, and z) as integers.
+2. Assign a value of 9 to x
+3. Assign a value of 7 to y
+4. Make z equal to the sum of those two variables. 
+5. Print out the contents of the z variable.
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:16]:
```suggestion
5. Print out the contents of the variable **z**.
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+
+### variables_2.py - Exercise 2
+Create a new file called calculations.txt And answer the following questions:
```
Comment (shiramax) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:19]:
```suggestion
Create a new file  **calculations.txt** and answer the following questions:
```

**Review by shiramax — CHANGES_REQUESTED** [review]

File: docs/course_materials/exercises/03_Variables_and_datatypes/class_exercise_2.md (hunk 7 above)
Comment (sleviim) [reply docs/course_materials/exercises/03_Variables_and_datatypes/class_exercise_2.md:9]: maybe car_name ?

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+
+For answering the following questions use the python terminal and run the commands you need.
+Finally, give a short answer to each question in calculation.txt file 
+
+1. What is the result of 10 ** 3?
+2. Given (x = 1), what will be the value of after we run (x += 2)?
+3. What is the result of float(1)?
+4. What is the result of 10 == “10”?
+5. Print the result of the following variable: 
+6. Number = ((((13 * 8 - 4) * 2 + 50) * 4 ) % 127 ) *5
```
Comment (sleviim) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:29]:
```suggestion
   Number = ((((13 * 8 - 4) * 2 + 50) * 4 ) % 127 ) *5     
```

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+ 
+ 
+### variables_3.py - Exercise 3
+1. Accept two numbers from the user (using input) and calculate multiplication (*)
+2. Accept two numbers from the user (using input) and calculate Subtract (-)
+3. Accept two numbers from the user (using input) and calculate Divide (/)
+4. Accept two numbers from the user (using input) and calculate Modulus (%) 
+ 
+
+
```
Comment (sleviim) [inline docs/course_materials/exercises/03_Variables_and_datatypes/homework_Variables.md:39]: redundant

File: docs/course_materials/variables_data_types.html (hunk 1 above)
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:80]:
```suggestion

```

File: docs/course_materials/variables_data_types.html (hunk 2 above)
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:83]:
```suggestion

```

File: docs/course_materials/variables_data_types.html [hunk 19]
```diff
@@ -39,12 +39,75 @@
...
+    >>> Hello Tom!
+    >>> print('Hello %s %s!' % (name, family_name))
+    >>> Hello Tom Sawyer!
+                    </code></pre>
+                </section>
+                <section>
+                <section data-markdown="exercises/03_Variables_and_datatypes/class_exercise_1.md" data-separator="^\n\n\n" data-separator-vertical="^\n\n">
+                </section>
+            </section>
+            </section>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:84]: Is this one redundant too?
Please make sure you can see the main rose background on the first presentation slide

File: docs/course_materials/variables_data_types.html [hunk 20]
```diff
@@ -39,12 +39,75 @@
...
                 <section>
                     <h1><b>
                             <span>Variables</span>
                     </h1></b>
                 </section>
+                <section>
+                    <h3><b>What is a Variable?</b></h3>
+                    <ul>
+                        <li>In computer programming, a variable has a name and contains a value.</li>
+                        <li>In most programming languages, a variable can store a single type of value. The kind of value that a variable can hold is also called a data type. </li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:95]:
```suggestion
                        <li>In most programming languages, a variable can store a single type of value</li>
                        <li>The kind of value that a variable can hold is also called a data type</li>
```

File: docs/course_materials/variables_data_types.html [hunk 21]
```diff
@@ -39,12 +39,75 @@
...
+                    </ul>
+                </section>
+                <section>
+                    <h3><b>Variables Names</b></h3>
+                    <ul>
+                        <li>A variable can have a short name (like a and b) or a more descriptive name (For example: age, carname, total_volume)</li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:101]:
```suggestion
                        <li>A variable can have a short name (like a and b) or a more descriptive name<br>
                        For example: age, carname, total_volume</li>
```

File: docs/course_materials/variables_data_types.html [hunk 22]
```diff
@@ -39,12 +39,75 @@
...
+                        <li>Rules for Python variables:</li>
+                          <ul>
+                            <li>A variable name must start with a letter or the underscore character</li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:104]:
```suggestion
                            <li>A variable name <b>must</b> start with a letter or the underscore character</li>
```

File: docs/course_materials/variables_data_types.html [hunk 23]
```diff
@@ -39,12 +39,75 @@
...
+                            <li>A variable name cannot start with a number</li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:105]:
```suggestion
                            <li>A variable name <b>cannot</b> start with a number</li>
```

File: docs/course_materials/variables_data_types.html [hunk 24]
```diff
@@ -39,12 +39,75 @@
...
+                            <li>A variable name can only contain alpha-numeric characters and underscores (A-z, 0-9, and _ )</li>
+                            <li>Variable names are case-sensitive (age, Age and AGE are three different variables)</li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:107]:
```suggestion
                            <li>Variable names are <b>case-sensitive</b> (age, Age and AGE are three different variables)</li>
```

File: docs/course_materials/variables_data_types.html [hunk 25]
```diff
@@ -130,14 +193,39 @@ <h3><b>Multiple Assignments</b></h3>
...
                 <section>
                     <h1><b>
-                            <span>Arithmetic Operators</span>
+                            <span>Numbers</span>
                     </h1></b>
                 </section>
+                <section>
+                    <h3><b>Number data types</b></h3>
+                    <ul>
+                        <li>Python supports integers, floating point numbers.</li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:206]:
```suggestion
                        <li>Python supports integers and floating point numbers</li>
```

File: docs/course_materials/variables_data_types.html [hunk 26]
```diff
@@ -130,14 +193,39 @@ <h3><b>Multiple Assignments</b></h3>
...
+                        <li>Integers and floating points are separated by the presence or absence of a decimal point:<br>5 is integer whereas 5.0 is a floating point number.</li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:207]:
```suggestion
                        <li>Integers and floating points are separated by the presence or absence of a decimal point:<br>
                        5 is an integer whereas 5.0 is a floating-point number</li>
```

File: docs/course_materials/variables_data_types.html [hunk 27]
```diff
@@ -130,14 +193,39 @@ <h3><b>Multiple Assignments</b></h3>
...
+                    </ul>
+                </section>
+                <section>
+                    <h3><b>Type conversion</b></h3>
+                    <ul>
+                        <li>Operations between a float and an integer will result in a float</li>
+                        <li>We can use functions like int() and float() to convert between types.<br>These functions can convert also from strings.</li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:214]:
```suggestion
                        <li>We can use functions like int() and float() to convert between types.<br>
                        These functions can convert also from strings</li>
```

File: docs/course_materials/variables_data_types.html [hunk 28]
```diff
@@ -39,12 +39,75 @@
...
+                <section>
+                    <h3><b>Interactive Input</b></h3>
+                    <ul>
+                        <li>In order to get an input:</li>
+                        <ul>
+                            <li>input()</li>
+                            <li>input(prompt) : prints prompt before reading input</li>
+                        </ul>
+                    </ul>
+                    <pre><code data-trim data-noescape class="hljs">
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:57]:
```suggestion
                    <pre><code data-trim data-noescape class="hljs lang-python">
```

File: docs/course_materials/variables_data_types.html [hunk 29]
```diff
@@ -39,12 +39,75 @@
...
+
+>>> address = input('Enter your address: ')
+Enter your address: 'Tel Aviv'
+>>> print(address)
+Tel Aviv
+                    </code></pre>
+                </section>
+                <section>
+                    <h3><b>Formatting</b></h3>
+                    <pre><code data-trim data-noescape class="hljs">
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:71]:
```suggestion
                    <pre><code data-trim data-noescape class="hljs lang-python"">
```

File: docs/course_materials/variables_data_types.html [hunk 30]
```diff
@@ -130,14 +193,39 @@ <h3><b>Multiple Assignments</b></h3>
...
+                        <li>Integers and floating points are separated by the presence or absence of a decimal point:<br>5 is integer whereas 5.0 is a floating point number.</li>
+                    </ul>
+                </section>
+                <section>
+                    <h3><b>Type conversion</b></h3>
+                    <ul>
+                        <li>Operations between a float and an integer will result in a float</li>
+                        <li>We can use functions like int() and float() to convert between types.<br>These functions can convert also from strings.</li>
+                    </ul>
+                    <pre><code data-trim data-noescape class="hljs">
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:216]:
```suggestion
                    <pre><code data-trim data-noescape class="hljs lang-python">
```

File: docs/course_materials/variables_data_types.html [hunk 31]
```diff
@@ -307,6 +361,14 @@ <h1><b>
                             <span>Strings</span>
                     </h1></b>
                 </section>
+                <section>
+                    <h3><b>What is String?</b></h3>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:365]:
```suggestion
                    <h3><b>What is a String?</b></h3>
```

File: docs/course_materials/variables_data_types.html [hunk 32]
```diff
@@ -307,6 +361,14 @@ <h1><b>
...
+                    <ul>
+                        <li>A string is a data type used in programming, such as an integer and float</li>
+                        <li>Represent text. For example, the word "hamburger" and the phrase "I ate 3 hamburgers" are both strings</li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:368]:
```suggestion
                        <li>Represents text. For example, the word "hamburger" and the phrase "I ate 3 hamburgers" are both strings</li>
```

File: docs/course_materials/variables_data_types.html [hunk 33]
```diff
@@ -307,6 +361,14 @@ <h1><b>
...
+                        <li>Even "12345" could be considered a string, if specified correctly. </li>
```
Comment (sleviim) [inline docs/course_materials/variables_data_types.html:369]:
```suggestion
                        <li>Even "12345" could be considered as a string, if specified correctly</li>
```

**Review by sleviim — CHANGES_REQUESTED** [review]
Please change all occurences of "class="hljs" to class="hljs lang-python" on the whole presentation

File: docs/course_materials/exercises/03_Variables_and_datatypes/homework_Strings.md [hunk 34]
```diff
@@ -1,41 +1,5 @@
 # Variables and Data Types - HomeWork exercise
```
[the remark this answers is not in the document]
Comment (sfishbai) [reply docs/course_materials/exercises/03_Variables_and_datatypes/homework_Strings.md:1]: Yes. there are 2 files for Variables and data type homeworks :
homework_exercise_1 - without String exercises
homework_exercise_2 - with String exercises

**Review by sleviim — APPROVED** [review]

**Pull request merged**